extensions/mongodb/pom.xml

```diff
@@ -48,12 +48,55 @@
                 </configuration>
             </plugin>
             <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
+                <version>${maven.shade.plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>fat-shaded-jar</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <filters>
+                                <filter>
+                                    <artifact>*:*</artifact>
+                                    <excludes>
+                                        <exclude>module-info.class</exclude>
+                                        <exclude>META-INF/*.SF</exclude>
+                                        <exclude>META-INF/*.DSA</exclude>
+                                        <exclude>META-INF/*.RSA</exclude>
+                                    </excludes>
+                                </filter>
+                            </filters>
+                            <relocations>
+                                <relocation>
+                                    <pattern>org.apache.calcite</pattern>
+                                    <shadedPattern>${relocation.root}.org.apache.calcite</shadedPattern>
```

Comment:
do we include Calcite in fat jar? will this be another copy of these classes apart from that from `hazelcast-sql`. `jar-with-dependencies` execution inherited from parent pom also includes it.

Reply:
Calcite has also some transitive dependencies like guava which are also used in the API

Reply:
Yes. The reason is simple: in hazelcast-sql we shade Calcite, so it's unusable outside hazelcast-sql. If we shade calcite in mongodb module as well, we can trick ClassLoader to think it's the same class (doesn't matter that in two modules)

Reply:
I tried to find information how maven-shade-plugin handles optional dependencies, but did not find anything. Is only has option for scope=provided. When build locally, fat jar contains too much.

Reply:
I excluded Calcite from shaded dependencies.

Reply:
how about other transitive dependencies, eg. guava and others?

Reply:
The jar should have the correct contents now - only `hazelcast-jet-mongodb` and mongo driver dependencies, no `hazelcast-sql`, calcite or its deps.